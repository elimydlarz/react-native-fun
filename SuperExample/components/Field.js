import React from 'react';
import { View, Text, TextInput } from 'react-native';

export default ({ updater, label, value, inputStyle }) => (
  <View>
    <Text style={{ fontSize: 18 }}>{label}</Text>
    <TextInput
      style={{ height: 50, borderWidth: 1, borderColor: 'black', ...inputStyle }}
      onChangeText={updater}
      value={value}
    />
  </View>
);
