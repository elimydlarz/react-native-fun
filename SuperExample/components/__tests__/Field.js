import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import Field from '../Field';

test('Field', () => {
  const tree = renderer.create(
    <Field
      updater={() => {}}
      label="label"
      value="value"
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Field label', () => {
  const tree = renderer.create(
    <Field
      updater={() => {}}
      label="label"
      value="value"
    />,
  ).toJSON();
  const subject = tree.children.find(c => c.type === 'Text').children[0];

  expect(subject).toEqual('label');
});

test('Field updater', () => {
  const updater = jest.fn();
  const tree = renderer.create(
    <Field
      updater={updater}
      label="label"
      value="value"
    />,
  ).toJSON();
  const onChangeText = tree.children.find(c => c.type === 'TextInput').props.onChangeText;

  onChangeText('bananas');

  expect(updater).toBeCalledWith('bananas');
});
