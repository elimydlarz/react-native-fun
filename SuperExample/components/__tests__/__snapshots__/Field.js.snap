// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
      }
    }
  >
    label
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "height": 50,
      }
    }
    value="value"
  />
</View>
`;
